File: python_pkg/python_pkg/class_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class PythonNode(Node):

    def __init__(self):
        super().__init__("py_node")
        self.get_logger().info("Hello ROS 2")
        self.counter = 0
        self.create_timer(0.5, self.timer_callback)
    
    def timer_callback(self):
        self.counter += 1
        self.get_logger().info("Hello: " + str(self.counter))

def main(args=None):
    rclpy.init(args=args)
    node = PythonNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: cpp_pkg/src/class_node.cpp
#include "rclcpp/rclcpp.hpp"

class CppNode:  public rclcpp::Node 
{
public:
    CppNode(): Node("class_cpp_node"){
        RCLCPP_INFO(this->get_logger(),"Hello C++ Node");
        timer = this->create_wall_timer(std::chrono::seconds(1),
            std::bind(&CppNode::timerCallback, this)
        );

    }
    ~CppNode(){};

private:

    void timerCallback(){
        counter++;
        RCLCPP_INFO_STREAM(this->get_logger(), "Hello: " << counter);
    }

    rclcpp::TimerBase::SharedPtr timer;
    int counter = 0;

};



int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CppNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}